draft-js-mention-plugin/src/index.js

```diff
@@ -82,54 +31,22 @@ const createMentionPlugin = (config = {}) => {
   const {
     mentionPrefix = '',
     theme = defaultTheme,
-    positionSuggestions = defaultPositionSuggestions,
   } = config;
-  const mentionSearchProps = {
-    ariaProps,
-    callbacks,
-    theme,
-    store,
-    entityMutability: config.entityMutability ? config.entityMutability : 'SEGMENTED',
-    positionSuggestions,
-  };
-  return {
-    MentionSuggestions: decorateComponentWithProps(MentionSuggestions, mentionSearchProps),
-    decorators: [
-      {
-        strategy: mentionStrategy,
-        component: decorateComponentWithProps(Mention, { theme, mentionPrefix }),
-      },
-      {
-        strategy: mentionSuggestionsStrategy,
-        component: decorateComponentWithProps(MentionSuggestionsPortal, { store }),
-      },
-    ],
-    getAccessibilityProps: () => (
-      {
-        role: 'combobox',
-        ariaAutoComplete: 'list',
-        ariaHasPopup: ariaProps.ariaHasPopup,
-        ariaExpanded: ariaProps.ariaExpanded,
-        ariaActiveDescendantID: ariaProps.ariaActiveDescendantID,
-        ariaOwneeID: ariaProps.ariaOwneeID,
-      }
-    ),
-
-    initialize: ({ getEditorState, setEditorState }) => {
-      store.getEditorState = getEditorState;
-      store.setEditorState = setEditorState;
-    },
-
-    onDownArrow: (keyboardEvent) => callbacks.onDownArrow && callbacks.onDownArrow(keyboardEvent),
-    onTab: (keyboardEvent) => callbacks.onTab && callbacks.onTab(keyboardEvent),
-    onUpArrow: (keyboardEvent) => callbacks.onUpArrow && callbacks.onUpArrow(keyboardEvent),
-    onEscape: (keyboardEvent) => callbacks.onEscape && callbacks.onEscape(keyboardEvent),
-    handleReturn: (keyboardEvent) => callbacks.handleReturn && callbacks.handleReturn(keyboardEvent),
-    onChange: (editorState) => {
-      if (callbacks.onChange) return callbacks.onChange(editorState);
-      return editorState;
-    },
+  const completionPlugin = createCompletionPlugin(
```

Comment:
The mentions plugin as well as Emoji currently take some parameters. Maybe I missed it, but it seems they are not passed down.

Instead of passing in multiple parameters I recommend to design the API in a way to pass in one object containing all sorts of configurations. This makes it easier to extend the config in the future and especially to eventually remove things.